ui/pages/confirmations/components/confirm/title/title.tsx

```diff
@@ -51,28 +52,34 @@ function ConfirmBannerAlert({ ownerId }: { ownerId: string }) {
 
 const ConfirmTitle: React.FC = memo(() => {
   const t = useI18nContext();
-  const currentConfirmation = useSelector(
-    currentConfirmationSelector,
-  ) as Confirmation;
+  const currentConfirmation = useSelector(currentConfirmationSelector);
 
   const typeToTitleTKey: Partial<Record<TransactionType, string>> = useMemo(
```

Comment:
This appears to have previously been just constant data to associate transaction types with titles.

Given it never changed, could it have originally been a traditional constant in the file instead of using a `useMemo` since there were no dependencies so we could have just returned the key itself, rather than resolving it via `t`.

As we're now making this more complex, by dynamically changing one of the values in the map based on the confirmation, would it be simpler to just create a top level `getTitle` function that accepts the current confirmation and `t` and then returns a string?

This would let us encapsulate it away and use whatever logic we like (switches for example) to generate the title?

Reply:
Done 👍